silo/api.py

```diff
@@ -106,7 +108,12 @@ def data(self, request, id):
 
         silo = Silo.objects.get(pk=id)
         if silo.public == False:
-            return HttpResponse("This table is not public. You must use the private API.")
+            url = reverse('silos-data', kwargs={'id': silo.pk})
+            params = '?query={}&shown_cols={}'\
```

Comment:
Do you know that you can use the function `urlencode` from the lib `urllib` with the `request.GET` as a parameter to create the URL query?